packages/actor-extract-links-extract-tree/lib/LinkOperator.ts

```diff
@@ -0,0 +1,31 @@
+import { v4 as uuidv4 } from 'uuid';
+import type { LogicOperator } from './solverInterfaces';
+/**
+ * A class representing a LogicOperation with an unique ID.
+ */
+export class LinkOperator {
+  /**
+     * The logic operator
+     */
+  public readonly operator: LogicOperator;
+  /**
+     * The unique ID
+     */
+  public readonly id: string;
+
+  /**
+     * Build a new LinkOperator with an unique ID.
+     * @param operator
+     */
+  public constructor(operator: LogicOperator) {
+    this.operator = operator;
+    this.id = uuidv4();
```

Comment:
Do we really need this?
Usage of UUID's typically indicates an architectural flaw.

Reply:
I think I might be able to do without it, it was more useful for my previous approach.